src/extras/PMREMGenerator.js

```diff
@@ -0,0 +1,180 @@
+                        uv.y *= -1.0;\
+                        if (faceIndex==0) {\
+                            sampleDirection = vec3(1.0, -uv.y, -uv.x);\
+                        }\
+                        else if (faceIndex==1) {\
+                            sampleDirection = vec3(-1.0, -uv.y, uv.x);\
+                        } else if (faceIndex==2) {\
+                            sampleDirection = vec3(uv.x, 1.0, uv.y);\
+                        } else if (faceIndex==3) {\
+                            sampleDirection = vec3(uv.x, -1.0, -uv.y);\
+                        } else if (faceIndex==4) {\
+                            sampleDirection = vec3(uv.x, -uv.y, 1.0);\
+                        } else {\
+                            sampleDirection = vec3(-uv.x, -uv.y, -1.0);\
+                        }\
+                        mat3 vecSpace = matrixFromVector(normalize(sampleDirection));\
+                        vec3 color = vec3(0.0);\
+                        const int NumSamples = 256;\
+                        vec3 vect;\
+                        float weight = 0.0;\
+                        for(int i=0; i<NumSamples; i++) {\
+                            float sini = sin(float(i));\
+                            float cosi = cos(float(i));\
+                            float rand = rnd(vec2(sini, cosi));\
+                            float blinnExp = GGXRoughnessToBlinnExponent(roughness);\
+                            vect = ImportanceSampleGGX(vec2(float(i) / float(NumSamples), rand), vecSpace, roughness);\
+                            float dotProd = dot(vect, normalize(sampleDirection));\
+                            weight += dotProd;\
+                            color += textureCube(sourceTexture, vect).rgb * dotProd;\
+                        }\
+                        color /= float(weight);\
+                        gl_FragColor = vec4( color, 1.0);\
+                   }";
+
+var materialPMREM = new THREE.ShaderMaterial(
+        {
+	uniforms: {
+		"faceIndex": { type: 'i', value: 0 },
+		"roughness": { type: 'f', value: 0.5 },
+		"sourceTexture": { type: 't', value: null }
+	},
+	vertexShader: vertexShaderPMREM,
+	fragmentShader: fragmentShaderPMREM
+        }
+);
+
+var PMREMGenerator = function( cubeTexture, resolution, numLods ) {
+
+	this.sourceTexture = cubeTexture;
+	this.cubeLods = [];
+	this.numSamplesPerLod = [];
+	this.numLods = numLods;
+	var size = resolution;
+	for ( var i = 0; i < this.numLods; i ++ ) {
+
+		var renderTarget = new THREE.WebGLRenderTargetCube( size, size,
+		{ format: THREE.RGBFormat, magFilter: THREE.LinearFilter, minFilter: THREE.LinearFilter } );
+		renderTarget.texture.generateMipmaps = false;
+		//renderTarget.texture.wrapS = THREE.ClampToEdgeWrapping;
+		//renderTarget.texture.wrapT = THREE.ClampToEdgeWrapping;
+		this.cubeLods.push( renderTarget );
+		if ( size > 16 )
+		size /= 2;
+
+	}
+
+	this.camera = new THREE.OrthographicCamera( - 1, 1, - 1, 1, - 0.01, 1000 );
```

Comment:
this.camera = new THREE.OrthographicCamera( - 1, 1, 1, - 1, 0, 1000 );

Is the camera inverted on purpose? Did you compensate somewhere else?

Also, it is not necessary, but it avoids confusion to keep the frustum in front of the camera, so the near plane distance should be >= 0.